### PR TITLE
DOC-4176:  Update IDM configuration documentation

### PR DESCRIPTION
**Pull request by mzorai**

### CODE REVIEW

File: docs/LDAP.rst [hunk 1]
```diff
@@ -16,6 +16,10 @@ LDAP Integration
 
     If SSO is what you need, read the :ref:`SSO chapter, eXo Add-ons guide <#eXoAddonsGuide.SSO>` that explains how eXo Platform works with
     LDAP through an SSO service like CAS or OpenAM.
+    
+.. warning:: -  eXo Platform supports only the **readonly** mode on LDAP/MSAD integration.
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
.. warning:: -  eXo Platform supports only the **read-only** mode on LDAP/MSAD integration.   
```

File: docs/LDAP.rst [hunk 2]
```diff
@@ -16,6 +16,10 @@ LDAP Integration
 
     If SSO is what you need, read the :ref:`SSO chapter, eXo Add-ons guide <#eXoAddonsGuide.SSO>` that explains how eXo Platform works with
     LDAP through an SSO service like CAS or OpenAM.
+    
+.. warning:: -  eXo Platform supports only the **readonly** mode on LDAP/MSAD integration.   
+             -  Only one single LDAP/MSAD is allowed.
+             -  The IDM mapped entities from LDAP/MSAD are imported in one way direction: **from LDAP to eXo Platform**.
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
             -  The IDM mapped entities from LDAP/MSAD are imported in one way direction: **from LDAP/MSAD to eXo Platform**. 
```

File: docs/LDAP.rst [hunk 3]
```diff
@@ -16,6 +16,10 @@ LDAP Integration
 
     If SSO is what you need, read the :ref:`SSO chapter, eXo Add-ons guide <#eXoAddonsGuide.SSO>` that explains how eXo Platform works with
     LDAP through an SSO service like CAS or OpenAM.
+    
+.. warning:: -  eXo Platform supports only the **readonly** mode on LDAP/MSAD integration.   
+             -  Only one single LDAP/MSAD is allowed.
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]: We should unify the term "LDAP/MSAD". Sometimes we use LDAP, sometimes LDAP/AD, sometimes LDAP/MSAD. I would go for LDAP/AD. WDYT ?

File: docs/LDAP.rst [hunk 4]
```diff
@@ -27,10 +31,6 @@ LDAP Integration
        A step by step tutorial for eXo Platform configuration with a
        directory server: LDAP/AD.
 
-    -  :ref:`Synchronization <LDAP.Synchronization>`
-       If you have integrated eXo Platform with a populated LDAP directory,
-       or when you manage users and groups via LDAP utilities, a service
-       to synchronize between them is necessary.
 
     -  :ref:`LDAP integration using legacy organization
        services <LDAP.LegacyOrganizationService>`
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]: These legacy services are not supported anymore, we should remove it from the documentation (I let you choose if you prefer to do it in a dedicated PR or in this one).

File: docs/LDAP.rst [hunk 5]
```diff
@@ -97,13 +97,6 @@ easy for readers, this tutorial is divided into four generic cases:
           represents users who are created via eXo Platform UI. The 
           understanding is similar to "*LDAP groups*\ " and "*Platform 
           groups*\ ".
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]: The following refs (below the note) should be removed since their content is removed:
* Platform users mapped into LDAP
* Platform groups mapped into LDAP

File: docs/LDAP.rst [hunk 6]
```diff
@@ -1737,10 +1173,9 @@ For the 5.1 version, this is the list of the supported ones:
 
 -  OpenLDAP.
 
--  Active directory: eXo Platform intregration with an Active directory 
-   is only supported in **readOnly**.
+-  Microsoft Active Directory.
 
--  OpenDJ.
+-  OpenDS.
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]: Only OpenLDAP and MS AD are supported (at least since PLF 4.4).

File: docs/LDAP.rst [hunk 7]
```diff
@@ -52,7 +52,7 @@ framework <https://www.jboss.org/picketlink/IDM>`__ that allows a very
 flexible integration with an LDAP directory:
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]: We should rename this chapter IMO, this can let think that there is another way to integrate with LDAP/AD.
Actually I think that all sub-chapter (Quick start, Configuration review, ...) must be directly put under the main chapter "LDAP Integration" and the content of "LDAP integration using PicketLink" should be moved in a sibling chapter simply called "PicketLink".
WDYT ?

File: docs/LDAP.rst [hunk 8]
```diff
@@ -16,6 +16,10 @@ LDAP Integration
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]: We should remove "The fact that an identity which is created via eXo Platform UI can be saved into the LDAP directory differentiates this integration with SSO use cases." since it is not true anymore.

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

File: docs/LDAP.rst (hunk 3 above)
Comment (plamarque) [reply docs/LDAP.rst]: why not simplifying and saying "directory" instead ?

File: docs/LDAP.rst (hunk 4 above)
Comment (mzorai) [reply docs/LDAP.rst]: in this one

File: docs/LDAP.rst (hunk 7 above)
Comment (mzorai) [reply docs/LDAP.rst]: OK

File: docs/LDAP.rst (hunk 3 above)
Comment (thomasdelhomenie) [reply docs/LDAP.rst]: Sounds good to me.

File: docs/LDAP.rst [hunk 9]
```diff
@@ -2723,7 +1442,7 @@ To resolve this, add an ACL (Access Control List) rule in the
     access to * by * read
 
 For more information, refer to `this
-discussion <https://community.exoplatform.com/portal/intranet/forum/topic/topicaf29ef7ca772acc44f16ba9a66b047bf>`__
+discussion <https://commPlatform users mapped into LDAPunity.exoplatform.com/portal/intranet/forum/topic/topicaf29ef7ca772acc44f16ba9a66b047bf>`__
```
Comment (vsellier) [inline docs/LDAP.rst]:
```suggestion
discussion <https://community.exoplatform.com/portal/intranet/forum/topic/topicaf29ef7ca772acc44f16ba9a66b047bf>`__
```

**Review by vsellier — DISMISSED** [review]

File: docs/LDAP.rst [hunk 10]
```diff
@@ -4,253 +4,165 @@
...
-          groups*\ ".
-		  The PicketLink IDM framework does not distinguish between
-		  LDAP-to-Platform and Platform-to-LDAP mapping, so the 
-		  configuration is basically the same, but the effect of some 
-		  parameters can be different. For example, the 
-		  ``createEntryAttributeValues`` parameter has no effect on the 
-		  LDAP-to-Platform mapping, thus is explained only in the 
-		  Platform-to-LDAP mapping.
+-  The supported directory implementations are: OpenLDAP and Microsoft Active Directory. You can refer to our official 
+   `supported environments <https://www.exoplatform.com/terms-conditions/supported-environments.pdf>`__ file for more
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
   `supported environments <https://www.exoplatform.com/terms-conditions/supported-environments.pdf>`__ matrix for more 
```

File: docs/LDAP.rst [hunk 11]
```diff
@@ -4,253 +4,165 @@
...
-   read-only or read-write mode. The LDAP directory can contain users
-   and groups, or only users.
-
--  It can be plugged into an empty LDAP directory.
-
--  Structure of users and groups in the LDAP directory can be finely
-   customized.
-
--  Multiple directories can be used as single datasources for eXo Platform.
+       How to resolve some possible issues of a directory integration.
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst:48]: There is no more "Frequently asked questions" chapter.

File: docs/LDAP.rst [hunk 12]
```diff
@@ -374,2358 +219,650 @@ So you can use the JDK built-in tool **jar**, as follows:
...
+Under the context DN (ou=Groups,o=acme,dc=example,dc=com), there are several groups as shown in the diagram below: 
+
+|image2|
+
+
+In this case, you should, in addition to previous steps described in the :ref:`Quick start section <LDAP.QuickStart>`, 
+follow these steps:
+
+1. Open the configuration file ``picketlink-idm-ldap-config.xml``.
+2. Search for the option ctxDNs to define the different locations of DNs
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
2. Search for the option ctxDNs to define the multiple locations of DNs
```

File: docs/LDAP.rst [hunk 13]
```diff
@@ -374,2358 +219,650 @@ So you can use the JDK built-in tool **jar**, as follows:
...
+   
+   - **idAttributeName**:  attribute name to use as the group id.
+   - **ctxDNs**: base DN of the groups in the directory.
+   - **entrySearchFilter**: search expression to filter objects to consider as groups.
+   - **parentMembershipAttributeName**: attribute which holds the list of group members. In OpenLDAP or MSAD default schemas, 
+     the member attribute is used, but your schema may use another attribute.
 
-In ``idm-configuration.xml``, the whole configuration is of eXo service.
-The eXo service configuration is started by either:
+Then this new object type must be referenced it in the PortalRepository repository:
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
Then this new object type must be referenced in the PortalRepository repository: 
```

File: docs/LDAP.rst [hunk 14]
```diff
@@ -374,2358 +219,650 @@ So you can use the JDK built-in tool **jar**, as follows:
...
-           <key>the_FQN_of_the_service_interface</key>
-           <type>the_FQN_of_the_service_implementation</type>
+.. _eXoConfiguration:
 
--  Or an external-component-plugin tag that looks like the following:
+2. **eXo configuration**
 
-   .. code:: xml
+   Besides the :ref:`PicketLink configuration <PicketlinkConfiguration>`, 
+   the eXo service configuration define in the file ``idm-configuration.xml`` must be updated.
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
   the eXo service configuration defined in the file ``idm-configuration.xml`` must be updated. 
```

File: docs/LDAP.rst [hunk 15]
```diff
@@ -374,2358 +219,650 @@ So you can use the JDK built-in tool **jar**, as follows:
...
+.. _eXoConfiguration:
 
--  Or an external-component-plugin tag that looks like the following:
+2. **eXo configuration**
 
-   .. code:: xml
+   Besides the :ref:`PicketLink configuration <PicketlinkConfiguration>`, 
+   the eXo service configuration define in the file ``idm-configuration.xml`` must be updated. 
+   A new entry must be added in the fields ``groupTypeMappings`` and ``ignoreMappedMembershipTypeGroupList`` 
+   to map the group defined in PicketLink configuration with the eXo Platform, as follows:
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
   to map the group defined in PicketLink configuration with the eXo Platform group, as follows: 
```

File: docs/LDAP.rst [hunk 16]
```diff
@@ -374,2358 +219,650 @@ So you can use the JDK built-in tool **jar**, as follows:
...
+of type ``org.exoplatform.services.organization.idm.Config``:
+
+-  **rootGroupName** : the name of the PicketLink IDM Group that will be 
+   used as a root parent. The default is ``GTN_ROOT_GROUP``.
+   
+-  **defaultGroupType**: the name of the PicketLink IDM GroupType that 
+   will be used to store groups. The default is ``GTN_GROUP_TYPE``.
+   
+-  **groupTypeMappings** : this parameter maps groups added with GateIn 
+   3.5 API as children of a given group ID, and stores them with a given
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
   API as children of a given group ID, and stores them with a given 
```

File: docs/LDAP.rst [hunk 17]
```diff
@@ -374,2358 +219,650 @@ So you can use the JDK built-in tool **jar**, as follows:
...
+This service has the following parameters defined as fields of ``object-param`` 
+of type ``org.exoplatform.services.organization.idm.Config``:
+
+-  **rootGroupName** : the name of the PicketLink IDM Group that will be 
+   used as a root parent. The default is ``GTN_ROOT_GROUP``.
+   
+-  **defaultGroupType**: the name of the PicketLink IDM GroupType that 
+   will be used to store groups. The default is ``GTN_GROUP_TYPE``.
+   
+-  **groupTypeMappings** : this parameter maps groups added with GateIn
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
-  **groupTypeMappings** : this parameter maps groups added with eXo Platform 
```

File: docs/LDAP.rst [hunk 18]
```diff
@@ -374,2358 +219,650 @@ So you can use the JDK built-in tool **jar**, as follows:
...
+   will be used to store groups. The default is ``GTN_GROUP_TYPE``.
+   
+-  **groupTypeMappings** : this parameter maps groups added with GateIn 
+   3.5 API as children of a given group ID, and stores them with a given 
+   group type name in PicketLink IDM.
+   If the parent ID ends with "/*", all child groups will have the mapped 
+   group type. Otherwise, only direct (first level) children will use this type.
+   This can be leveraged by LDAP if the LDAP DN is configured in PicketLink 
+   IDM to only store a specific group type. This will then store the given 
+   branch in the GateIn 3.5 group tree, while all other groups will remain in the database.
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
   branch in the eXo Platform group tree, while all other groups will remain in the database.
```

File: docs/LDAP.rst [hunk 19]
```diff
@@ -374,2358 +219,650 @@ So you can use the JDK built-in tool **jar**, as follows:
...
+   
+-  **forceMembershipOfMappedTypes**: groups stored in PicketLink IDM with 
+   a type mapped in 'groupTypeMappings' will automatically be members under the mapped parent. 
+   The Group relationships linked by the PicketLink IDM group association will not be necessary.
+   This parameter can be set to false if all groups are added via GateIn 3.5 APIs. This may be 
+   useful with the LDAP configuration when being set to true, it will make every entry added to 
+   LDAP appear in GateIn 3.5. This, however, is not true for entries added via GateIn 3.5 management UI.
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
   LDAP appear in eXo Platform. This, however, is not true for entries added via eXo Platform management UI.
```

File: docs/LDAP.rst [hunk 20]
```diff
@@ -374,2358 +219,650 @@ So you can use the JDK built-in tool **jar**, as follows:
...
+   If the parent ID ends with "/*", all child groups will have the mapped 
+   group type. Otherwise, only direct (first level) children will use this type.
+   This can be leveraged by LDAP if the LDAP DN is configured in PicketLink 
+   IDM to only store a specific group type. This will then store the given 
+   branch in the GateIn 3.5 group tree, while all other groups will remain in the database.
+   
+-  **forceMembershipOfMappedTypes**: groups stored in PicketLink IDM with 
+   a type mapped in 'groupTypeMappings' will automatically be members under the mapped parent. 
+   The Group relationships linked by the PicketLink IDM group association will not be necessary.
+   This parameter can be set to false if all groups are added via GateIn 3.5 APIs. This may be
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
   This parameter can be set to false if all groups are added via eXo Platform APIs. This may be 
```

File: docs/LDAP.rst [hunk 21]
```diff
@@ -374,2358 +219,650 @@ So you can use the JDK built-in tool **jar**, as follows:
...
+-  **associationMembershipType** : if this option is used, each Membership created with MembrshipType that 
+   is equal to the value specified here, will be stored in PicketLink IDM as the simple Group-User association.
+
+-  **passwordAsAttribute**: this parameter specifies if a password should be stored using the PicketLink IDM 
+   Credential object or as a plain attribute. The default value is set to false.
+   
+-  **useParentIdAsGroupType**: this parameter stores the parent ID path as a group type in PicketLink IDM 
+   for any IDs not mapped with a specific type in 'groupTypeMappings'. If this option is set to false, 
+   and no mappings are provided under 'groupTypeMappings', only one group with the given name can exist 
+   in the GateIn 3.5 group tree.
```
Comment (thomasdelhomenie) [inline docs/LDAP.rst]:
```suggestion
   in the eXo Platform group tree.
```

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]

**Review by thomasdelhomenie — APPROVED** [review]
